cmake/rosconsole-extras.cmake.in

```diff
@@ -13,3 +13,11 @@ if("@ROSCONSOLE_BACKEND@" STREQUAL "log4cxx")
 endif()
 
 cmake_policy(POP)
+
+if("@ROSCONSOLE_BACKEND@" STREQUAL "log4cxx" AND "@LOG4CXX_REQUIRES_CXX17@")
+  if(NOT DEFINED CMAKE_CXX_STANDARD OR "${CMAKE_CXX_STANDARD}" LESS 17)
+    message(STATUS "rosconsole is upgrading C++ standard to C++17")
+    set(CMAKE_CXX_STANDARD 17)
```

Comment:
It is unfortunate that the `rosconsole_log4cxx` target isn't exported, but this is a clever workaround. It looks like this should be compatible with Noetic on 20.04, so with a little testing I think this is ok to merge.

Reply:
Yes this is meant to do nothing on Ubuntu 20.04, so it should not change anything there.